internal/api/download.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+)
+
+func (c *ApiClient) Download(ctx Context, reportType string, reportJournalType string, reportScheduleType string, reportAccountType string, reportDebtType string, dateField string, dateFromField string, dateToField string, emailField string) error {
```

Comment:
`Field` is redundant naming here. Also, `dateField` needs to be more descriptive to distinguish it from the other date fields.